pennylane/qchem/convert.py

```diff
@@ -746,14 +657,124 @@ def _rcisd_state(cisd_solver, tol=1e-15):
     # alpha, beta -> alpha, beta excitations
     # generate all possible pairwise combinations of _single_ excitations of alpha and beta sectors
     rowvals, colvals = np.array(list(product(c1a_configs, c1a_configs)), dtype=int).T
-    c2ab = (c2.transpose(0, 2, 1, 3).reshape(nocc * nvir, -1)).ravel()
+    c2ab = c2.transpose(0, 2, 1, 3).reshape(nocc * nvir, -1)
     dict_fcimatr.update(
-        dict(zip(list(zip(rowvals, colvals)), c2ab * np.kron(c1a_signs, c1a_signs)))
+        dict(
+            zip(
+                list(zip(rowvals, colvals)),
```

Comment:
At first glance, it feels like this is unnecessary. You might be able to get the same thing by directly using `list(product(c1a_configs, c1a_configs)` (or something similar), which will help you skip defining `rowvals` and `colvals`.

Reply:
Good catch -- we transpose and separate it only to zip it back again. Fixed.